src/org/zaproxy/zap/extension/fuzz/httpfuzzer/processors/tagcreator/HttpFuzzerMessageProcessorTagUIPanel.java

```diff
@@ -0,0 +1,200 @@
+        JLabel label = new JLabel(labelName,  JLabel.TRAILING);
+        fieldsPanel.add(label);
+        label.setLabelFor(textField);
+        fieldsPanel.add(textField);
+    }
+
+    @Override
+    public JPanel getComponent() {
+        return fieldsPanel;
+    }
+
+    @Override
+    public void setFuzzerMessageProcessorUI(HttpFuzzerMessageProcessorTagUI processorUI) {
+        TagRule rule = processorUI.getTagRule();
+        if(rule.getClass() == MatchByRegexTagRule.class){
+            MatchByRegexTagRule tagRule = (MatchByRegexTagRule)rule;
+            matchByRegexRadio.setSelected(true);
+            setTextAndDiscardAllEdits(matchByRegexFieldRegex, tagRule.getRegex());
+            setTextAndDiscardAllEdits(matchByRegexFieldTag, tagRule.getTag());
+        }
+        else if(rule.getClass() == ExtractByRegexTagRule.class){
+            ExtractByRegexTagRule tagRule = (ExtractByRegexTagRule)rule;
+            extractByRegexRadio.setSelected(true);
+            setTextAndDiscardAllEdits(extractByRegexFieldRegex, tagRule.getRegex());
+        }
+    }
+
+    private void setTextAndDiscardAllEdits(ZapTextField zapTextField, String text){
+        zapTextField.setText(text);
+        zapTextField.discardAllEdits();
+    }
+
+    @Override
+    public HttpFuzzerMessageProcessorTagUI getFuzzerMessageProcessorUI() {
+        TagRule tagRule = null;
+        if(matchByRegexRadio.isSelected()){
+            tagRule = new MatchByRegexTagRule(
+                    matchByRegexFieldRegex.getText(),
+                    matchByRegexFieldTag.getText());
+
+        } else if(extractByRegexRadio.isSelected()){
+            tagRule = new ExtractByRegexTagRule(extractByRegexFieldRegex.getText());
+        }
+        return new HttpFuzzerMessageProcessorTagUI(tagRule);
+    }
+
+    @Override
+    public void clear() {
+        matchByRegexRadio.setSelected(true);
+        setTextAndDiscardAllEdits(matchByRegexFieldRegex, "");
+        setTextAndDiscardAllEdits(matchByRegexFieldTag, "");
+        setTextAndDiscardAllEdits(extractByRegexFieldRegex, "");
+    }
+
+    @Override
+    public boolean validate() {
+        if(matchByRegexRadio.isSelected()){
+            return validateMatchByRegex();
+        } else if(extractByRegexRadio.isSelected()){
+            return validateExtractByRegex();
+        }
+        return INVALID;
+    }
+
+    private boolean validateMatchByRegex() {
+        return isTextFieldNotEmpty(matchByRegexFieldTag) &&
+                isTextFieldNotEmpty(matchByRegexFieldRegex) &&
+                hasTextFieldValidRegex(matchByRegexFieldRegex);
+    }
+
+    private boolean hasTextFieldValidRegex(ZapTextField zapTextField){
+        String regex = zapTextField.getText();
+        try {
+            Pattern.compile(regex);
+        } catch (PatternSyntaxException exception) {
+            String errorMessage = MessageFormat.format(VALIDATION_REGEX_SYNTAX_ERROR, exception.getDescription());
```

Comment:
Although the description of the regex error is useful it's not internationalised.